lib/DriverTool/swift_parse_test_main.cpp

```diff
@@ -0,0 +1,235 @@
+    std::error_code err;
+    for (auto I = llvm::sys::fs::directory_iterator(path, err),
+              E = llvm::sys::fs::directory_iterator();
+         I != E; I.increment(err)) {
+      _loadSwiftFilesRecursively(I->path(), buffers);
+    }
+  } else if (path.endswith(".swift")) {
+    if (auto buffer = llvm::MemoryBuffer::getFile(path)) {
+      buffers.push_back(std::move(*buffer));
+    }
+  }
+}
+
+/// Load all '.swift' files in the specified \p filePaths into \p buffers.
+/// If the path is a directory, this recursively collects the files in it.
+static void
+loadSources(ArrayRef<std::string> filePaths,
+            SmallVectorImpl<std::unique_ptr<llvm::MemoryBuffer>> &buffers) {
+  for (auto path : filePaths) {
+    _loadSwiftFilesRecursively(path, buffers);
+  }
+}
+
+/// Measure the duration of \p body execution.
+template <typename Duration>
+static std::pair<Duration, Duration> measure(llvm::function_ref<void()> body) {
+  auto cStart = std::clock();
+  auto tStart = std::chrono::steady_clock::now();
+  body();
+  auto cEnd = std::clock();
+  auto tEnd = std::chrono::steady_clock::now();
+
+  auto clockMultiply =
+      Duration::period::den / CLOCKS_PER_SEC / Duration::period::num;
+
+  Duration cDuration((cEnd - cStart) * clockMultiply);
+  return {std::chrono::duration_cast<Duration>(tEnd - tStart),
+          std::chrono::duration_cast<Duration>(cDuration)};
+}
+
+/// Perform the performance measurement using \c Executor .
+/// Parse all \p buffers using \c Executor , \p iteration times, and print out
+/// the measurement to the \c stdout.
+template <typename Executor>
+static void
+perform(const SmallVectorImpl<std::unique_ptr<llvm::MemoryBuffer>> &buffers,
+        ExecuteOptions opts, unsigned iteration, uintmax_t totalBytes,
+        uintmax_t totalLines) {
+
+  llvm::outs() << "----\n";
+  llvm::outs() << "parser: " << Executor::name << "\n";
+
+  using duration_t = std::chrono::nanoseconds;
+  auto tDuration = duration_t::zero();
+  auto cDuration = duration_t::zero();
+
+  for (unsigned i = 0; i < iteration; i++) {
+    for (auto &buffer : buffers) {
+      std::pair<duration_t, duration_t> elapsed = measure<duration_t>(
+          [&] { Executor::performParse(buffer->getMemBufferRef(), opts); });
+      tDuration += elapsed.first;
+      cDuration += elapsed.second;
+    }
+  }
+
+  auto tDisplay =
+      std::chrono::duration_cast<std::chrono::milliseconds>(tDuration).count();
+  auto cDisplay =
+      std::chrono::duration_cast<std::chrono::milliseconds>(cDuration).count();
+
+  auto byteTPS = totalBytes * duration_t::period::den / cDuration.count();
+  auto lineTPS = totalLines * duration_t::period::den / cDuration.count();
+
+  llvm::outs() << llvm::format("wall clock time (ms): %8d\n", tDisplay)
+               << llvm::format("cpu time (ms):        %8d\n", cDisplay)
+               << llvm::format("throughput (byte/s):  %8d\n", byteTPS)
+               << llvm::format("throughput (line/s):  %8d\n", lineTPS);
```

Comment:
Would it make sense to specify that throughput is measured in CPU time, not wall time?